src/transformers/trainer.py

```diff
@@ -2462,12 +2396,49 @@ def _save_checkpoint(self, model, trial, metrics=None):
         else:
             torch.save(rng_states, os.path.join(output_dir, f"rng_state_{self.args.process_index}.pth"))
 
-        if self.args.push_to_hub:
-            self._push_from_checkpoint(output_dir)
+    def _save_optimizer_and_scheduler(self, output_dir):
```

Comment:
Hmm, it would be great if we could reuse accelerate code here, but it seems that there is no directly corresponding code. I wonder if we could achieve that with a bit of refactoring on the accelerate side @muellerzr (potentially in a future PR).

Reply:
I looked into this the last time during migration to Accelerate backend as well and recently, I couldn't find a clean way of doing it on the Accelerate side. Maybe a discussion on this offline would help.